### ISSUE TITLE
[4.2] Cannot publish package configuration unless it is in the root of the package

### ISSUE THREAD

**Issue by tpavlek** [laravel/framework#7243]
If you have a package `foo/bar` and you store the configuration file in `vendor/foo/bar/src/FrameworkIntegration/Laravel/config/config.php`, it is not possible to publish this configuration with `php artisan config:publish`.

Relevant source is in line 85 of `\Illuminate\Foundation\ConfigPublisher`:

```php
	protected function getSource($package, $packagePath)
	{
		$source = $packagePath."/{$package}/src/config";

		if ( ! $this->files->isDirectory($source))
		{
			throw new \InvalidArgumentException("Configuration not found.");
		}

		return $source;
	}
```

The source is hardcoded to `/src/config` in the package folder. This should be updated to pull the location of the config from the service provider which should have registered the location of the assets.

**Comment by GrahamCampbell** [comment]
You can override this by passing the path you want as the third param to the package method.

**Comment by tpavlek** [comment]
@GrahamCampbell No, unless I'm missing something I don't believe you can?

Passing the third param will load the package configuration properly when in use via \Config::get(), etc. but the ConfigPublisher seems hardcoded to never touch the path registered with the ServiceProvider.

I created an example laravel application here: https://github.com/tpavlek/test-pkg-config.

It contains only the stock laravel installation, plus my package oauth2-twitch. If you try to run `php artisan config:publish depotwarehouse/oauth2-twitch` you get Configuration not found.

Debugging through the registration of the Service Provider, it is properly set via `$this->app['config']->package` in `\Illuminate\Support\ServiceProvider` on line 2991. However, the behaviour of the ConfigPublisher is exactly as I described in the bug report.

**Comment by GrahamCampbell** [comment]
Hmmm, well this is already fixed in 5.0 anyway, and there are no plans to make any changes to 4.2's package system. I think 5.0 is much more flexible that 4.2 was both for packages, and for some other things too (pagination for example).

**Comment by tpavlek** [comment]
Absolutely.

I plan to migrate as much as possible to 5.0 as soon as it's drop and I'm sure it's not going to break production, but I spent my morning debugging this issue so thought I'd make the tracker aware of it.
